CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 - Fix not logging OpenAPI queries when `log-query=main-query` is enabled by @steve-chavez in #4226
 - Fix not logging explain query when `log-query=main-query` is enabled by @steve-chavez in #4319
+- Fix not logging transaction variables and db-pre-request function when `log-query=main-query` is enabled by @steve-chavez in #3934
```

Comment:
Ah, I understood that the `main-query` only included the [definition mentioned in the docs](https://docs.postgrest.org/en/v12/references/transactions.html#main-query) (as mentioned in https://github.com/PostgREST/postgrest/pull/3904#issuecomment-2666684939). Not against it, but it confused me a bit at the beginning.

Reply:
Right, so I'm thinking that we should just have one `log-query = true` as originally planned. I believe we have some coercion on the configs, so `log-query = "main-query"` will become `true`.

Why do this:

- More log-query values add extra effort when testing all the possibilities.
- Complicates things for the users and us maintainers.
- Originally we added `main-query` because of a limitation of the previous implementation.

WDYT?

Reply:
Even if it's not a backwards compatible change it's worth it IMO.

It's more beneficial overall to put the effort into logging every SQL query including the [schema cache queries ](https://github.com/PostgREST/postgrest/issues/3934#issuecomment-2702367462) than to worry about testing these multiple config values.

Reply:
Yes, that was my initial idea too, but was just limited by the scope in that PR. So I agree, let's go with that.